app/src/main/java/com/zulip/android/models/Message.java

```diff
@@ -104,7 +105,7 @@
     private String subject;
     @SerializedName("timestamp")
     @DatabaseField(columnName = TIMESTAMP_FIELD)
-    private Date timestamp;
+    private long timestamp;
```

Comment:
As column field is changed so DB version is increased and onUpdate method is called. Thus it working for old messages too. :)